pallets/gear/src/lib.rs

```diff
@@ -1242,9 +1243,27 @@ pub mod pallet {
 
             let (builtins, _) = T::BuiltinDispatcherFactory::create();
             let ext_manager = ExtManager::<T>::new(builtins);
-            ext_manager.set_program(packet.destination(), &code_info, message_id, block_number);
 
             let program_id = packet.destination();
+
+            // Before storing the program to `ProgramStorage` we need to make sure that an account
+            // can be created for the program.
+            // Note: making a transfer outside of the `Ext::set_program()` because here a transfer
+            // is allowed to fail (as opposed to creating a program by a program).
+            CurrencyOf::<T>::transfer(
+                &who,
+                &program_id.cast(),
+                CurrencyOf::<T>::minimum_balance(),
+                ExistenceRequirement::KeepAlive,
```

Comment:
why keep alive?
sender is free to be deleted actually

Reply:
makes sense